lib/mirage_configure.ml

```diff
@@ -29,97 +30,6 @@ let clean_myocamlbuild () =
 
 let opam_path ~name = Fpath.(v name + "opam")
 
-let artifact ~name = function
-  | #Mirage_key.mode_solo5 as tgt ->
-      let ext = snd (Mirage_configure_solo5.solo5_pkg tgt) in
-      let file = Fpath.v (name ^ ext) in
-      (file, file)
-  | #Mirage_key.mode_unix ->
-      (Fpath.((v "_build" / "main") + "native"), Fpath.v name)
-  | #Mirage_key.mode_xen ->
-      let file = Fpath.(v name + "xen") in
-      (file, file)
-
-let additional_artifacts ~name =
-  let libvirt = Mirage_configure_libvirt.filename ~name in
-  function
-  | `Xen -> Fpath.[ v name + "xl"; v name + "xl.in"; v name + "xe"; libvirt ]
-  | `Virtio -> [ libvirt ]
-  | _ -> []
-
-let find_git () =
-  let is_git p = Bos.OS.Dir.exists Fpath.(p / ".git") in
-  let app_opt p d = match p with None -> d | Some p -> Fpath.(d // p) in
-  let rec find p path =
-    if Fpath.is_root p then Error (`Msg "no git repo found, reached root")
-    else
-      is_git p >>= fun has_git ->
-      if has_git then Ok path
-      else find (Fpath.parent p) (Some (app_opt path (Fpath.base p)))
-  in
-  Bos.OS.Dir.current () >>= fun cwd ->
-  find cwd None >>= fun subdir ->
-  let git_branch = Bos.Cmd.(v "git" % "rev-parse" % "--abbrev-ref" % "HEAD") in
-  Bos.OS.Cmd.(run_out git_branch |> out_string) >>= fun (branch, _) ->
-  let git_remote = Bos.Cmd.(v "git" % "remote" % "get-url" % "origin") in
-  Bos.OS.Cmd.(run_out git_remote |> out_string) >>| fun (git_url, _) ->
-  (subdir, branch, git_url)
-
-let configure_opam ~name info =
-  let open Codegen in
-  let subdir, git_info =
-    match find_git () with
-    | Error _ -> (None, None)
-    | Ok (subdir, branch, git_url) -> (subdir, Some (branch, git_url))
-  in
-  let file = opam_path ~name in
-  with_output file
-    (fun oc () ->
-      let fmt = Format.formatter_of_out_channel oc in
-      append fmt "# %s" (generated_header ());
-      Info.pp_opam ~name fmt info;
-      append fmt {|maintainer: "dummy"|};
-      append fmt {|authors: "dummy"|};
-      append fmt {|homepage: "dummy"|};
-      append fmt {|bug-reports: "dummy"|};
-      Format.fprintf fmt {|build: [ "sh" "-exc" "|};
-      ( match subdir with
-      | None -> ()
-      | Some p -> Format.fprintf fmt "cd %a && " Fpath.pp p );
```

Comment:
This is very ugly and fragile, so I haven't copied that in my PR.  Instead I'm using the `--config-file` argument here. Not sure if that's make a difference, I'll try to test this (not sure how yet...)

Reply:
Hum actually I think I understand what you tried to do here -- I'll try to replicate and somehow test that behaviour automatically.